pallets/pooled-staking/src/lib.rs

```diff
@@ -754,4 +755,42 @@ pub mod pallet {
             pools::distribute_rewards::<T>(&candidate, rewards)
         }
     }
+    impl<T: Config> tp_traits::StakingCandidateHelper<Candidate<T>> for Pallet<T> {
+        fn is_candidate_selected(candidate: &Candidate<T>) -> bool {
+            <SortedEligibleCandidates<T>>::get()
+                .into_iter()
+                .any(|c| &c.candidate == candidate)
+        }
+        fn on_online_status_change(
+            candidate: &Candidate<T>,
+            _is_online: bool,
+        ) -> DispatchResultWithPostInfo {
+            Calls::<T>::update_candidate_position(&[candidate.clone()])
```

Comment:
in my opinion this is where you should mark collators as offline or not.  you need to make sure that pallet-staking filters out non-eligible collators (because they are offline). So I think it makes sense to have a fucntion here called sorted_eligible_candidates where you filter out those offline, hence the storage item needs to be in this pallet (the extrinsic though, can be in another pallet). Or you create a struct here that applies the filder from inactivity pallet.